src/Aspire.Hosting/ResourceBuilderExtensions.cs

```diff
@@ -706,6 +709,67 @@ public static IResourceBuilder<T> WithHealthCheck<T>(this IResourceBuilder<T> bu
         return builder;
     }
 
+    /// <summary>
+    /// Adds a health check to the resource which is mapped to a specific endpoint.
+    /// </summary>
+    /// <typeparam name="T">A resource type that implements <see cref="IResourceWithEndpoints" />.</typeparam>
+    /// <param name="builder">A resource builder.</param>
+    /// <param name="path">The relative path to test.</param>
+    /// <param name="statusCode">The result code to interpret as healthy.</param>
```

Comment:
Would be good to document the defaults on this and other parameters as applicable.